libpod/runtime_ctr.go

```diff
@@ -246,6 +249,40 @@ func (r *Runtime) newContainer(ctx context.Context, rSpec *spec.Spec, options ..
 }
 
 func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Container, retErr error) {
+	if ctr.IsDefaultInfra() || ctr.IsService() {
```

Comment:
setupContainer() container is run once when the container is created, as such the logic here will not work once you reboot as the tmpfs dir location will no longer be there.

I am pretty sure we must have this in generateSpec() code path in libpod.